retroshare-gui/src/gui/gxsforums/CreateGxsForumMsg.cpp

```diff
@@ -568,11 +568,43 @@ void CreateGxsForumMsg::fileHashingFinished(QList<HashedFile> hashedFiles)
 	QList<HashedFile>::iterator it;
 	for (it = hashedFiles.begin(); it != hashedFiles.end(); ++it) {
 		HashedFile& hashedFile = *it;
+		QString ext = QFileInfo(hashedFile.filename).suffix().toUpper();
+
 		RetroShareLink link = RetroShareLink::createFile(hashedFile.filename, hashedFile.size,
 		                                                 QString::fromStdString(hashedFile.hash.toStdString()));
-		if (link.valid()) {
-			mesgString += link.toHtmlSize() + "<br>";
+
+		if (hashedFile.flag & HashedFile::Picture) {
+			mesgString += QString("<img src=\"file:///%1\" width=\"100\" height=\"100\">").arg(hashedFile.filepath);
+			mesgString += "<br>";
+		} else {
+			bool preview = false;
+			if(hashedFiles.size()==1 && (ext == "JPG" || ext == "PNG" || ext == "JPEG" || ext == "GIF" || ext == "WEBP"))
+			{
+				QString encodedImage;
+				uint32_t maxMessageSize = 32000;
+				if (RsHtml::makeEmbeddedImage(hashedFile.filepath, encodedImage, 640*480, maxMessageSize - 200 - link.toHtmlSize().length()))
```

Comment:
Use MAX_ALLOWED_GXS_MESSAGE_SIZE as in line 555

Reply:
but that is to big, i need to make small thumbnails, 

> static const uint32_t MAX_ALLOWED_GXS_MESSAGE_SIZE = 199000;

Reply:
else i can use this :

> maxMessageSize = rsMsgs->getMaxMessageSecuritySize(RS_CHAT_TYPE_LOBBY);

Reply:
Why don't change 640*480 so?
And don't call it maxMessageSize but maxImageSize.

Reply:
oh i forget this 

> Why don't change 640*480 so? And don't call it maxMessageSize but maxImageSize.